packages/astro/test/astro-get-static-paths.test.js

```diff
@@ -4,19 +4,57 @@ import { loadFixture } from './test-utils.js';
 describe('getStaticPaths()', () => {
 	let fixture;
 
-	before(async () => {
-		fixture = await loadFixture({
-			projectRoot: './fixtures/astro-get-static-paths/',
-			buildOptions: {
-				site: 'https://mysite.dev/blog/',
-				sitemap: false,
-			},
+	describe('number of build calls', () => {
+		before(async () => {
+			fixture = await loadFixture({
+				projectRoot: './fixtures/astro-get-static-paths/',
+				buildOptions: {
+					site: 'https://mysite.dev/blog/',
+					sitemap: false,
+				},
+			});
+			await fixture.build();
+		});
+		it('is only once during build', () => {
+			// useless expect; if build() throws in setup then this test fails
+			expect(true).to.equal(true);
 		});
-		await fixture.build();
 	});
 
-	it('is only called once during build', () => {
-		// useless expect; if build() throws in setup then this test fails
-		expect(true).to.equal(true);
+	describe('dynamic', () => {
+		let fixture;
+		let devServer;
+
+		before(async () => {
+			fixture = await loadFixture({ projectRoot: './fixtures/astro-get-static-paths/' });
+			devServer = await fixture.startDevServer();
+		});
+
+		after(async () => {
+			devServer && devServer.stop();
+		});
+		describe('[plain]', () => {
+			it('resolves 200 on matching static path', async () => {
+				const res = await fixture.fetch('/pizza/provolone-sausage');
+				expect(res.status).to.equal(200);
+			});
+
+			it('resolves 404 on bad static path', async () => {
+				const res = await fixture.fetch('/pizza/provolone-pineapple');
+				expect(res.status).to.equal(404);
+			});
+		});
+
+		describe('[...spread]', () => {
+			it('resolves 200 on matching static path', async () => {
+				const res = await fixture.fetch('/pizza/grimaldis/new-york');
+				expect(res.status).to.equal(200);
+			});
+
+			it('resolves 404 on bad static path', async () => {
+				const res = await fixture.fetch('/pizza/pizza-hut');
+				expect(res.status).to.equal(404);
+			});
+		});
```

Comment:
Nice tests! Only feedback would be that it's not clear what `dynamic`, `[plain]`, and `[...spread]` mean without looking at the fixture.

Reply:
Ah, was thinking about this from the Jest mindset of flattening. Aka this becomes `getStaticPaths - dynamic - [plain]`. Is there a term I can use for `[this]` and `[...this]`?

Reply:
Ah that makes sense. We tend to not nest `describe` and just do it all at the top level.

But `[plain]` would be a named parameter and `[...spread]` would be a rest parameter. Both under the category of "Dynamic routing"

Reply:
Ah gotcha. Was mostly nesting for shared `before` blocks as well

Reply:
Ah I was actually wrong. We usually do one top-level `describe` for the shared `before` block, then flat `it()` statements inside. 

See [`astro-expr.test.js`](https://github.com/withastro/astro/blob/main/packages/astro/test/astro-expr.test.js) for an example. If you wouldn't mind refactoring this, that'd be great!